src/Agent/NewRelic/Profiler/Configuration/InstrumentationConfiguration.h

```diff
@@ -45,52 +45,82 @@ namespace NewRelic { namespace Profiler { namespace Configuration
         }
 
         InstrumentationConfiguration(InstrumentationPointSetPtr instrumentationPoints) :
-            _instrumentationPoints(instrumentationPoints),
+            _instrumentationPointsMap(new InstrumentationPointMap()),
             _invalidFileCount(0)
-        {}
-
-        InstrumentationPointSetPtr GetInstrumentationPoints() {
-            return _instrumentationPoints;
+        {
+            for (auto instrumentationPoint : *instrumentationPoints)
+            {
+                (*_instrumentationPointsMap)[instrumentationPoint->GetMatchKey()] = instrumentationPoint;
+            }
         }
 
         uint16_t GetInvalidFileCount()
         {
             return _invalidFileCount;
         }
 
-        InstrumentationPointPtr TryGetInstrumentationPoint(const MethodRewriter::IFunctionPtr function) const
+        InstrumentationPointSetPtr GetInstrumentationPoints()
         {
-            SignatureParser::MethodSignaturePtr methodSignature = SignatureParser::SignatureParser::ParseMethodSignature(function->GetSignature()->begin(), function->GetSignature()->end());
-            InstrumentationPointPtr ipToFind(new InstrumentationPoint());
-            ipToFind->AssemblyName = function->GetAssemblyName();
-            ipToFind->ClassName = function->GetTypeName();
-            ipToFind->MethodName = function->GetFunctionName();
-            ipToFind->Parameters = std::unique_ptr<xstring_t>(new xstring_t(methodSignature->ToString(function->GetTokenResolver())));
+            auto result = InstrumentationPointSetPtr(new InstrumentationPointSet());
+            for (auto instrumentationPointKVP : *_instrumentationPointsMap) {
+                result->insert(instrumentationPointKVP.second);
+            }
+
+            return result;
```

Comment:
Maybe we should keep the original `_instrumentationPoints` field in addition to the map, especially if the map is just holding references to items in the `_instrumentationPoints` set? That would avoid the additional dynamic allocation occurring in this method, which I think is never getting deallocated.

Reply:
done